### ISSUE TITLE
Error Verifying Address On Ledger

### ISSUE THREAD

**Issue by mbrobert** [cryptoadvance/specter-desktop#1840]
**Describe the bug**
When attempting to display the address on the device, I get an error. Specter is able to detect the ledger via usb and the device/wallet is synced with specter, however, I cant seem to do anything with it.

**To Reproduce**
1. Go to Wallets
2. Click on Receive
3. Click Display address on device
4. Plug in Ledger, and select device when it appears on the screen.
5. Pop up says "verify address matches address shown on device" but then it closes and the error message appears.
6. 

**Expected behavior**

**Screenshots**

**Desktop**
 - Where is your node running: Citadel Node
 - Where is your specter running: On Desktop PC
 - OS: Windows
 - Browser: Brave Browser
 - Specter Version 1.10.5

Attached is a screenshot of the code and the txt file contains the logs
![SpectreDesktopErrorCode](https://user-images.githubusercontent.com/54902495/184894203-5e5ed52a-89c0-4253-b9a7-01a25c1b65e3.PNG)
[SpectreDesktopError.txt](https://github.com/cryptoadvance/specter-desktop/files/9351523/SpectreDesktopError.txt)

**Comment by elproducto** [comment]
Want to add same error  is also occurring when attempting to Sign transaction with a Ledger Nano S.

Where is your node running: Bitcoin Core 
Where is your specter running: Windows 10
OS: Windows 10 21H2 (Remote) + Debian 11.4.0(Node + Specter)
Browser: NA Specter Desktop on Windows
Specter Version 1.10.5

`2022-08-23T03:48:01.298Z [info] : stderr-Traceback (most recent call last):
  File "cryptoadvance\specter\util\json_rpc.py", line 35, in jsonrpc
  File "cryptoadvance\specter\helpers.py", line 81, in wrapper_fn
  File "cryptoadvance\specter\hwi_rpc.py", line 328, in sign_tx
  File "hwilib\commands.py", line 197, in signtx
  File "hwilib\devices\ledger.py", line 130, in func
  File "hwilib\devices\ledger.py", line 312, in sign_tx
  File "hwilib\devices\ledger_bitcoin\client.py", line 81, in register_wallet
hwilib.devices.ledger_bitcoin.exception.errors.IncorrectDataError: ('0x6a80', 'Error in <BitcoinInsType.REGISTER_WALLET: 2> command', '')`

![Untitled12](https://user-images.githubusercontent.com/3057562/186045185-9838b7db-5c9b-4882-8fdd-897b0a561321.png)

**Comment by k9ert** [comment]
I can't reproduce. I guess most important is the firmware versions. I'm not that familiar with Ledger but my one is running 
* secure element 1.6.1
* Microcontroller 1.12
* Bootloader 0.11
* Bitcoin 1.5.1

Can't reproduce the issue. Can you check/update your firmware?

**Comment by elproducto** [comment]
For reference my Ledger Nano S information below.  
@k9ert - If possible and you are using an Legder Nano S, can you update your ledger firmware and update the Bitcoin app installed?

Secure Element - 2.1.0
Microcontoller 1.12
Bootloader 0.11
Bitcoin Ledger App version - 2.0.6

**Comment by jeee** [comment]
I am having the same issue on a Nano X:
Secure element 2.0.2
Microcontroller 2.30
bootloader 1.16
Bitcoin Ledger App version - 2.0.6

./hwi.py enumerate
[{"type": "ledger", "model": "ledger_nano_x", "label": null, "path": "1-2:1.0", "fingerprint": "f30419f4", "needs_pin_sent": false, "needs_passphrase_sent": false}]

**Comment by elproducto** [comment]
Output of hwi enumerate below:

[{"type": "ledger", "model": "ledger_nano_s", "label": null, "path": "DevSrvsID:4295044056", "fingerprint": "Redacted", "needs_pin_sent": false, "needs_passphrase_sent": false}]

I also noticed the following error that appeared after installing HWI.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
cryptoadvance-specter 1.10.5 requires hwi==2.1.0, but you have hwi 2.1.1 which is incompatible.

**Comment by k9ert** [comment]
> I also noticed the following error that appeared after installing HWI.

You should definitely test the exact same version of HWI which is also used in specter. `v1.10.5` is already outdated. Can you update to the newest `v1.12.0`?

@jeee hwi version would be valuable as well. Which one is it? I don't have a Citadel and can only test the newest versions.

**Comment by elproducto** [comment]
Tested against v1.12.0.  Still not able to  sign tx.  I was able to successfully "Test connection", in USB Devices section. Output from log below.  


```
2022-09-13T01:52:55.209Z [info] : stderr-SPECTERD: ERROR in specter_error: Unexpected error:
SPECTERD: ERROR in specter_error: ----START-TRACEBACK-----------------------------------------------------------------

2022-09-13T01:52:55.209Z [info] : stderr-SPECTERD: ERROR in specter_error: Request error for method getreceivedbyaddress: Address not found in wallet
Traceback (most recent call last):
  File "cryptoadvance/specter/wallet.py", line 844, in getdata
  File "cryptoadvance/specter/rpc.py", line 450, in fn
cryptoadvance.specter.rpc.RpcError: Request error for method getreceivedbyaddress: Address not found in wallet

2022-09-13T01:52:55.210Z [info] : stderr-SPECTERD: ERROR in specter_error: ----END---TRACEBACK-----------------------------------------------------------------
SPECTERD: INFO in wallet: Didn't get transactions on change address bc1qh4av86h05zy2spquzp5x96kznvugtyksaqq3292cm47he2488cpsk70v93. Refilling keypool.

2022-09-13T01:53:19.155Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:19] "OPTIONS /hwi/api/ HTTP/1.1" 200 -

2022-09-13T01:53:20.067Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:20] "POST /hwi/api/ HTTP/1.1" 200 -

2022-09-13T01:53:45.751Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:45] "OPTIONS /hwi/api/ HTTP/1.1" 200 -

2022-09-13T01:53:46.650Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:46] "POST /hwi/api/ HTTP/1.1" 200 -

2022-09-13T01:53:48.817Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:48] "POST /hwi/api/ HTTP/1.1" 200 -

2022-09-13T01:53:58.808Z [info] : stderr-Traceback (most recent call last):
  File "cryptoadvance/specter/util/json_rpc.py", line 35, in jsonrpc
  File "cryptoadvance/specter/helpers.py", line 99, in wrapper_fn
  File "cryptoadvance/specter/hwi_rpc.py", line 328, in sign_tx
  File "hwilib/commands.py", line 197, in signtx
  File "hwilib/devices/ledger.py", line 130, in func
  File "hwilib/devices/ledger.py", line 312, in sign_tx
  File "hwilib/devices/ledger_bitcoin/client.py", line 81, in register_wallet
hwilib.devices.ledger_bitcoin.exception.errors.IncorrectDataError: ('0x6a80', 'Error in <BitcoinInsType.REGISTER_WALLET: 2> command', '')

2022-09-13T01:53:58.810Z [info] : stderr-SPECTERD: INFO in _internal: 127.0.0.1 - - [12/Sep/2022 21:53:58] "POST /hwi/api/ HTTP/1.1" 200 -

```

**Comment by k9ert** [comment]
Looks like this: https://github.com/bitcoin-core/HWI/issues/605

> The upcoming version of the Ledger bitcoin app (ETA 1-2 months) will correctly support SIGHASH_DEFAULT as well.

So, to me it looks like a solution would be an update of the Ledger Bitcoin App when this is released. A "workaround" would be changing that in [embit](https://github.com/diybitcoinhardware/embit) to address the special case for ledger but i'm not favoring such workarounds for bad firmware-implementations. 

However,feel free to create an issue there (and reference this issue and the one of HWI). If we have a new release of embit with that change, i'm happy to make a new specter-release within days.

**Comment by k9ert** [comment]
Referencing @bigspider here. This is not fixed by #1888, right?

**Comment by Tronics21** [comment]
Hello Specter Team

Thank you very much for your excellent work. I appreciate Specter Desktop, very much. I'm using v1.10.3 as it's the last version to support the Ledger Nano S. I reported the bug to Specter after the release of v1.10.5. Now v1.13.0 is already released and the error with the leather Nano S is still present. If you want to make a transaction and you have to confirm the transaction with the publisher, the error message comes (Internal error: (0x6a80',' Error in command',").) I hope they can fix the problem so I can also make an update against it.

**Comment by bigspider** [comment]
> Referencing @bigspider here. This is not fixed by #1888, right?

No, #1888 is only in preparation of the coming releases.
I notice now that Specter is shipping with hwi 2.1.0, and [hwi 2.1.1 solved a signing failure](https://github.com/bitcoin-core/HWI/releases/tag/2.1.1), so I'd expect that upgrading to 2.1.1 will fix this one.

Still some work to do to improve support (e.g.: show multisig addresses), will try to do a PR to hwi soon.

(_en passant_: The SIGHASH_DEFAULT issue mentioned is only a potential issue for taproot transactions)

**Comment by k9ert** [comment]
Thanks @bigspider . Somehow overlooked your comment but it was also a long weekend in germany. Very much appreciated!
I've created a new pre-release `v1.13.1-pre1` which should soon appear here in order to test the fix:
https://github.com/cryptoadvance/specter-desktop/releases

**Comment by k9ert** [comment]
We have a pre-release with this issue (hopefully) fixed. Can someone confirm that this is fixed?
https://github.com/cryptoadvance/specter-desktop/releases/tag/v1.13.1-pre1

**Comment by moneymanolis** [comment]
@bigspider I don't have a Ledger to test the [pre-release](https://github.com/cryptoadvance/specter-desktop/releases/tag/v1.13.1-pre1), if you have a moment, would be great if you could test it. Thanks!

**Comment by bigspider** [comment]
> @bigspider I don't have a Ledger to test the [pre-release](https://github.com/cryptoadvance/specter-desktop/releases/tag/v1.13.1-pre1), if you have a moment, would be great if you could test it. Thanks!

I tried the prerelease on:

Ledger Nano S 2.1.0
Bitcoin app version 2.0.6

and I was regularly able to sign a multisig 3-of-3 transaction.

On the other hand, I already used Specter in the past and it worked for me, so I'm not sure I'm in exactly the same conditions as the people who had issues above.

Hopefully @elproducto or @Tronics21 could try again?

**Comment by elproducto** [comment]
Confirmed working for me

**Comment by moneymanolis** [comment]
Thanks guys!

**Comment by lilianchisca** [comment]
is it possible to get the macos version for this prerelease as well? having the same issue

**Comment by moneymanolis** [comment]
> is it possible to get the macos version for this prerelease as well? having the same issue

Tagging @k9ert for that.

**Comment by k9ert** [comment]
Done, @lilianchisca !
https://github.com/cryptoadvance/specter-desktop/releases/tag/v1.13.1-pre1

**Comment by lilianchisca** [comment]
@k9ert that was fast! thank you mate!

**Comment by wombat6** [comment]
This PR is closed, but I wanted to add my own test results:

- I was able to reproduce the error alert with v1.13.0 using a Ledger Nano X. I tried to receive funds, and I told Specter Desktop to show the address on the device.

![error](https://user-images.githubusercontent.com/112285082/196122112-d5ff259e-e402-4fa4-a680-6e0b462e52e8.png)

- After upgrading to v1.13.1-pre1, I retried receiving the funds, and it worked.
